### DISCUSSION TITLE
Hosting online

### DISCUSSION THREAD

**Discussion by ypgrimm** [Selbi182/SpotifyBigPicture#21] [Q&A] [answered]
Is there a way to run the code online so that I don't need to keep the application open in the background and can access it from anywhere? For example, hosting it somewhere and being able to access it on my phone or laptop. Also is there a way to change the name of the tab that opens so it's just the song name.

**Comment by chalter-Innovativesol** [comment]
I host it as an ECS container service in my AWS account and have a record in my domain pointing to it. That way I can access it from anywhere and don't need to worry about having my laptop running this service all the time. Costs is negligible if you use the smallest deployment size.

**Reply by ypgrimm** [reply]
I managed to make a ec2 server and have the spotify full screen work on there but i cant access the localhost on any other device. I tried to change the firewall properties but no luck. Any ideas?

**Reply by Selbi182** [reply]
You don't need to access the localhost from the outside, but rather whatever IP and port you need to access the service from the outside. I don't know how AWS really works in that regard, but you gotta probably expose the port or something. I also recommend doing the login process locally on your computer, so you can simply upload the spotifybot.properties file with the tokens already there.

**Reply by ypgrimm** [reply]
I ended up hosting it on CloudFlare tunnels and hosting it on the ec2 but thank you for all your help!

**Comment by Selbi182** [comment, marked as answer]
To run the app from anywhere you'd need to find a hosting solution so you can access it like any web service. I want to rewrite the codebase eventually to at one point be browser-side only, but that'll take some time.

As for the tab thing, I don't quite understand what you're looking for. You mean an option to just display the song name without anything else?

**Reply by ypgrimm** [reply]
![image](https://github.com/Selbi182/SpotifyBigPicture/assets/78011816/2e7ebe32-f268-417d-ad6a-9e3734c262e8)
Is there a way to remove the "SpotifyBigPicture" part of the tab name?

**Reply by Selbi182** [reply]
I'll add that feature.

**Reply by ypgrimm** [reply]
Thanks 👍

**Reply by Selbi182** [reply]
New release with settings to customize the website title is now live: https://github.com/Selbi182/SpotifyBigPicture/releases/tag/v1.5

**Reply by ypgrimm** [reply]
Wow that was fast, thanks! Any idea why I don't have the update on some devices?

**Reply by Selbi182** [reply]
What do you mean?

**Reply by ypgrimm** [reply]
Never mind it just took a while to update on Firefox for some reason.

**Reply by Selbi182** [reply]
Probably a caching issue then. Ctrl+F5 usually does the trick as well. I need to figure out how to force a hard refresh on version updates.
